source/crud/crud.rst

```diff
@@ -1927,8 +1928,83 @@ The ``update`` family of operations require that the update document parameter M
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Historically, only primaries could execute an aggregation pipeline with ``$out``
+or ``$merge``. As of ``featureCompatibilityVersion`` 4.4, secondaries can now
+execute such pipelines; however, since drivers do not track
+``featureCompatibilityVersion``, executing an aggregation pipeline with ``$out``
+or ``$merge`` on a secondary will require a 5.0+ server (i.e. wire version >=
+13).
+
+Drivers MUST guarantee the following:
+
+- If an explicit (i.e. per-operation) read preference is specified for an
+  aggregation with a write stage, drivers MUST attempt to use it. If that would
+  result in a pre-5.0, secondary server being selected, drivers MUST instead
+  select a server using a primary read preference.
+- If no explicit read preference is specified but a default read preference is
+  available to inherit (e.g. from the Collection), drivers MUST attempt to use
+  it. If that would result in a pre-5.0, secondary server being selected,
+  drivers MUST instead select a server using a primary read preference.
+
+Drivers SHOULD use a custom server selector to consider server/wire version when
+matching a read preference and, if a pre-5.0 secondary would be selected, fall
+back to selecting a primary. With this approach, only a single server selection
+attempt is needed.
+
+If it is not possible to augment the server selector and a pre-5.0 secondary is
+selected, drivers MAY invoke server selection a second time to obtain a primary.
+However, if the first attempt results in a server selection timeout, drivers
+MUST NOT make a second attempt and MUST propagate the original timeout error. If
+the topology type is Single, Sharded, or LoadBalanced, drivers MUST NOT make a
+second attempt (there is no benefit in doing so).
+
+Regardless of whether drivers employ a custom server selector or two-attempt
+approach, they MUST discern the *effective* read preference for the operation,
+which SHALL be be used for specifying the
+`$readPreference global command argument <../message/OP_MSG.rst#global-command-arguments>`_
+and
+`passing read preference to mongos and load balancers <../server-selection/server-selection.rst#passing-read-preference-to-mongos-and-load-balancers>`_
+(if applicable). The effective read preference SHALL be discerned as follows:
+
+- In the absence of an explicit or inherited read preference, the effective read
+  preference is ``{ "mode": "primary" }``.
+- If an explicit or inherited read preference results in selection of an
+  ineligible server and selection falls back to a primary, the effective read
+  preference is ``{ "mode": "primary" }``.
+- If a pre-5.0 mongos or a load balancer backing a pre-5.0 mongos is selected,
+  the effective read preference is ``{ "mode": "primary" }``.
+- If an explicit or inherited read preference is used and results in selection
+  of an eligible server (excluding a pre-5.0 mongos as discussed above), that is
+  the effective read preference.
```

Comment:
@benjirewis: I rewrote the last paragraph and added this bulleted list to discern what the _effective_ read preference is for populating `$readPreference` in the command. Please let me know if this addresses your previous questions.

Reply:
Looks great to me! Thanks for this comprehensive list 🧑‍🔧